server/src/test/java/org/opensearch/cluster/routing/WeightedRoundRobinTests.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class WeightedRoundRobinTests extends OpenSearchTestCase {
```

Comment:
Lets add tests for below scenarios
1. 1:1:0
2. 0:0:0
3. -1:0:1
4. -1:3:0:10000

Reply:
Added tests